Automattic/jetpack
# Search: handle new product purchase flow

Fixes #25907 and #25908

#### Changes proposed in this Pull Request:
- Add connection support for new pricing page
- Add jetpack_search_free product flow

#### Other information:


#### Jetpack product discussion
n/a

#### Does this pull request change what data or activity we track or use?
no

#### Testing instructions:
* Create a fresh site with Search plugin but no Search subscription
* Make sure to do a `git pull` on your sandbox
* Add the following code to your wp-content/mu-plugins/0-sandbox.php file on you sandbox, creating the file if necessary
  * `define( 'USE_STORE_SANDBOX', true );`
* Sandbox your test site: `define( 'JETPACK__SANDBOX_DOMAIN', 'xxx.xxx.xxx.xxx.com' );`
* Open `/wp-admin/admin.php?page=jetpack-search&new_pricing_202208=1`
* Ensure the upsell page handles connection okay
* Ensure you could purchase paid and free products from the plan page - please refer to test instructions in `D88792-code` for how to pay with test credit cards in store sandbox

Known issues:
- The pricing is not yet right on the plan page, which would need `D88792-code` <- WIP
- After purchasing the free product, the redirection is not right, which is probably about some Calypso setup and will be addressed later.
